### PR TITLE
ref(store): Remove Kafka event publisher from `StoreView.process`

### PR DESCRIPTION
**Pull request by tkaemming**
This removes the Kafka event publishing block from the `StoreView.process` method as well as the options used to configure it. This doesn't affect the `APIView._publish_to_kafka` method that has been used for normalization testing.

It's possible we'll want to bring back Kafka publishing from this view later, but it'll happen at a different point in this process.

### CODE REVIEW

**Review by tkaemming — COMMENTED** [review]
Mostly just trying to reduce surface area for future changes.

**Review by mitsuhiko — APPROVED** [review]
lgtm. Just had to make sure the other kafka publisher for testing purposes survives. Don't remove that one.

File: tests/sentry/test_event_consumer.py [hunk 1]
```diff
@@ -1,34 +0,0 @@
...
-    def test_event_consumer(self, mock_kafka_publisher):
-        with self.options({
-            'store.kafka-sample-rate': 1.0,
-            'store.process-in-kafka': True,
-            'kafka-publisher.raw-event-sample-rate': 0.0,
-        }):
-            mock_event_accepted = mock.Mock()
-            event_accepted.connect(mock_event_accepted)
-
-            resp = self._postWithHeader({'logentry': {'message': u'hello'}})
```
Comment (tkaemming) [inline tests/sentry/test_event_consumer.py]: Didn't realize there was coupling in this test case here, so this also needs to be removed as part of this PR as well for a clean split.

**Comment by tkaemming** [comment]
> Just had to make sure the other kafka publisher for testing purposes survives. Don't remove that one.

Yep, wasn't planning on it.

**Pull request merged**